src/components/VHeader/VFilterButton.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+  <VButton
+    variant="action-menu"
+    class="self-center"
+    :pressed="sidebarVisible"
+    @click="toggleFilters"
+  >
+    <FilterIcon v-show="showIcon" class="w-6 h-6" width="24" height="24" />
+    <span class="filter-label">{{ buttonLabel }}</span>
+  </VButton>
+</template>
+
+<script>
+import FilterIcon from '~/assets/icons/filter.svg?inline'
+import { computed, defineComponent, useContext } from '@nuxtjs/composition-api'
+import { useMediaQuery } from '~/composables/use-media-query'
+import { SEARCH } from '~/constants/store-modules'
+import { SET_FILTER_IS_VISIBLE } from '~/constants/mutation-types'
+
+const VFilterButton = defineComponent({
+  name: 'FilterButton',
+  components: {
+    FilterIcon,
+  },
+  props: {
+    /**
+     * Whether the page has been scrolled. If true on mobile, the button
+     * label is blank.
+     * This is currently not implemented, and is always false.
+     * TODO(@obulat): add use-scroll composable to VHeader
+     * @default false
+     */
+    isScrolled: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props) {
+    const { i18n, store } = useContext()
+    const { isScrolled = false } = props
+
+    const defaultWindow = typeof window !== 'undefined' ? window : undefined
+    const isTablet = useMediaQuery('(min-width: 640px)', defaultWindow)
+
+    const appliedFilterTags = computed(
+      () => store.getters['search/appliedFilterTags']
+    )
+    const isAnyFilterApplied = computed(
+      () => appliedFilterTags.value.length > 0
+    )
+
+    /**
+     * On screens wider than tablet: 'Filters'.
+     * On mobile screens:
+     * 1. If no filters are applied - '' (icon only).
+     * 2. If the page is scrolled down - '' (icon only).
+     * 3. If the page wasn't scrolled, and some filters are applied:
+     * the number of applied filters, e.g. '7'.
+     *
+     * @type {ComputedRef<string>}
```

Comment:
For TS you'll need to explicitly import these types:

```suggestion
     * @type {import('@nuxtjs/composition-api').ComputedRef<string>}
```